packages/fxa-auth-server/scripts/stripe-products-and-plans-to-firestore-documents/stripe-product-and-plans-converter.ts

```diff
@@ -0,0 +1,344 @@
+      }
+    }
+    return urlConfig;
+  }
+
+  // TODO JSDoc
+  // This method will return a valid ProductConfig, but it will need to be
+  // appended with any locale data for Stripe plans if present.
+  stripeProductToProductConfig(product: Stripe.Product): ProductConfig {
+    // Firestore cannot serialize class instances
+    const productConfig: any = {};
+
+    // BaseConfig
+    productConfig.active = product.active;
+    productConfig.capabilities =
+      this.capabilitiesMetadataToCapabilityConfig(product);
+    // `locales` will be populated once we iterate through the Stripe Plans
+    productConfig.locales = {};
+    productConfig.styles = this.stylesMetadataToStyleConfig(product);
+    productConfig.support = this.supportMetadataToSupportConfig(product);
+    productConfig.uiContent = this.uiContentMetadataToUiContentConfig(product);
+    productConfig.urls = this.urlMetadataToUrlConfig(product);
+
+    // Extended by ProductConfig
+    if (product.id) {
+      productConfig.stripeProductId = product.id;
+    }
+    if (product.metadata.productSet) {
+      productConfig.productSet = product.metadata.productSet;
+    }
+    if (product.metadata.promotionCodes) {
+      productConfig.promotionCodes = commaSeparatedListToArray(
+        product.metadata.promotionCodes
+      );
+    }
+    return productConfig;
+  }
+
+  findLocaleStringFromStripePlan(plan: Stripe.Plan): null | string {
+    // Try to extract a locale from the plan nickname
+    const { nickname } = plan;
+    let locale = nickname
+      ?.split(' ')
+      .filter((w) => this.supportedLanguages.includes(w.toLowerCase()));
+    if (locale && locale.length > 0) {
+      return locale[0];
+    }
+    // Failing that, try to ascertain the language of any localized strings
+    //  - Use Google Translate Client https://cloud.google.com/translate/docs/samples/translate-v3-detect-language#translate_v3_detect_language-nodejs
+    // If the locale is not valid, throw an error
+    // Return the locale
+    return null;
+  }
+
+  // Extract any locale data from a Stripe plan and convert it to ProductConfig.locales
+  stripePlanLocalesToProductConfigLocales(
+    plan: Stripe.Plan
+  ): ProductConfig['locales'] {
+    const locales: ProductConfig['locales'] = {};
+    const localeStr = this.findLocaleStringFromStripePlan(plan);
+    if (!localeStr) {
+      return locales;
+    }
+    const uiContent = this.uiContentMetadataToUiContentConfig(plan);
+    const urls = this.urlMetadataToUrlConfig(plan);
+    const support = this.supportMetadataToSupportConfig(plan);
+    const locale: ProductConfig['locales'][string] = {
+      uiContent,
+      urls,
+      support,
+    };
+    locales[localeStr] = locale;
+    return locales;
+  }
+
+  // TODO JSDoc
+  stripePlanToPlanConfig(
+    plan: Stripe.Plan,
+    productConfigId: string
+  ): PlanConfig {
+    return {
+      stripePriceId: plan.id,
+      active: plan.active,
+      productConfigId,
+    };
```

Comment:
Since I'm commenting on a draft, I don't know if it's an error or not, but this function is not complete.

Reply:
Yeah I'm not done with this method; I should have mentioned it at the top or put a TODO comment there, sorry.